Use logger

martinsumner opened the pull request:
Update kv_index_tictcatree to use logger - by using exported functions from leveled_log module.

Also adds tag to metadata to support routing to different log file sin Riak

martinsumner committed:
Use logger

Update kv_index_tictcatree to use logger - by using exported functions from leveled_log module.

ThomasArts commented on src/aae_util.erl:
```diff
@@ -259,19 +312,6 @@ get_segmentid(B, K) ->
-    ok = log("G8888", [], [], [info, warn, error]),
-    SW = os:timestamp(),
-    ok = log_timer("G0001", [], SW, [], [warn, error]),
-    ok = log_timer("G8888", [], SW, [], [info, warn, error]),
```
This test would fail if you had adapted it? Hinting that updating these simple tests may have a bit of value anyway

martinsumner replied:
I want to keep tests of logging within leveled, where the log functions exist - this was a copy & paste test to test the copy & paste code which was duplicated within here previously.  I've added  an extra test to the leveled PR to make the behaviour more explicit within leveled_log module.

martinsumner committed:
Remove leveled_rand

There is an issue outstanding to resolve in leveled - https://github.com/martinsumner/leveled/issues/456
